File: examples/audit/benchmark-langs/python/bench.py
import time, math

def fib(n):
    return n if n < 2 else fib(n - 1) + fib(n - 2)

def cuadrado(x):
    return x * x

def now_ms():
    return time.perf_counter() * 1000.0

# arith
N = 20_000_000
t0 = now_ms()
s = 0
for i in range(N):
    s += i; s -= 1; s *= 2; s //= 2; s %= 1_000_000
    if s < 0: s = 0
t1 = now_ms()
print("arith_result:", s)
print("arith_ms:", round(t1 - t0, 3))

# fib
t0 = now_ms()
r = fib(30)
t1 = now_ms()
print("fib_result:", r)
print("fib_ms:", round(t1 - t0, 3))

# array
N = 100_000
t0 = now_ms()
arr = [i for i in range(N)]
asum = sum(arr)
t1 = now_ms()
print("arr_len:", len(arr))
print("arr_sum:", asum)
print("arr_ms:", round(t1 - t0, 3))

# string
N = 10_000
t0 = now_ms()
s = ""
for i in range(N):
    s += "x"
t1 = now_ms()
print("str_len:", len(s))
print("str_ms:", round(t1 - t0, 3))

# math
N = 200_000
t0 = now_ms()
acc = 0.0
for i in range(N):
    acc += math.sqrt(i + 1)
    acc += math.sin(i)
t1 = now_ms()
print("math_result:", acc)
print("math_ms:", round(t1 - t0, 3))

# calls
N = 1_000_000
t0 = now_ms()
csum = 0
for i in range(N):
    csum += cuadrado(i)
t1 = now_ms()
print("call_result:", csum)
print("call_ms:", round(t1 - t0, 3))

File: examples/audit/benchmark-langs/js/bench.js
"use strict";

function fib(n) {
    return n < 2 ? n : fib(n - 1) + fib(n - 2);
}
function cuadrado(x) {
    return x * x;
}
const now_ms = () => performance.now();

// arith
const N_ARITH = 20000000;
let t0 = now_ms();
let sum = 0;
for (let i = 0; i < N_ARITH; i++) {
    sum += i; sum -= 1; sum *= 2; sum = Math.floor(sum / 2); sum %= 1000000;
    if (sum < 0) sum = 0;
}
let t1 = now_ms();
console.log("arith_result:", sum);
console.log("arith_ms:", (t1 - t0).toFixed(3));

// fib
t0 = now_ms();
let r = fib(30);
t1 = now_ms();
console.log("fib_result:", r);
console.log("fib_ms:", (t1 - t0).toFixed(3));

// array
const N_ARR = 100000;
t0 = now_ms();
const arr = [];
for (let i = 0; i < N_ARR; i++) arr.push(i);
let asum = 0;
for (const x of arr) asum += x;
t1 = now_ms();
console.log("arr_len:", arr.length);
console.log("arr_sum:", asum);
console.log("arr_ms:", (t1 - t0).toFixed(3));

// string
const N_STR = 10000;
t0 = now_ms();
let s = "";
for (let i = 0; i < N_STR; i++) s += "x";
t1 = now_ms();
console.log("str_len:", s.length);
console.log("str_ms:", (t1 - t0).toFixed(3));

// math
const N_MATH = 200000;
t0 = now_ms();
let acc = 0.0;
for (let i = 0; i < N_MATH; i++) {
    acc += Math.sqrt(i + 1);
    acc += Math.sin(i);
}
t1 = now_ms();
console.log("math_result:", acc);
console.log("math_ms:", (t1 - t0).toFixed(3));

// calls
const N_CALL = 1000000;
t0 = now_ms();
let csum = 0;
for (let i = 0; i < N_CALL; i++) csum += cuadrado(i);
t1 = now_ms();
console.log("call_result:", csum);
console.log("call_ms:", (t1 - t0).toFixed(3));
